shepherd_utils/broker.py

```diff
@@ -112,26 +112,43 @@ async def acquire_lock(
     logger: logging.Logger,
 ):
     """Acquire a redis lock for a given row."""
+    got_lock = False
     try:
         client = await aioredis.Redis(
             connection_pool=lock_redis_pool,
         )
-        locked = await client.get(response_id)
-        if locked is None:
-            await client.setex(response_id, 45, consumer_id)
-            await client.aclose()
-            return True
-        for i in range(60):
-            await asyncio.sleep(5)
-            locked = await client.get(response_id)
-            if locked is None:
-                await client.setex(response_id, 45, consumer_id)
-                await client.aclose()
-                return True
-        return False
+        pubsub = client.pubsub()
+        await pubsub.subscribe(response_id)
+        for i in range(12):
+            aquired = await client.set(response_id, consumer_id, ex=45, nx=True)
+            if aquired:
+                got_lock = True
+                break
+            try:
+                await pubsub.get_message(ignore_subscribe_messages=True, timeout=5)
+            except asyncio.TimeoutError:
+                pass
+            # await asyncio.sleep(1)
+            # try again
+
+        await pubsub.unsubscribe(response_id)
+        await pubsub.aclose()
+        await client.aclose()
+        return got_lock
```

Comment:
In the next line, and at other exceptions, do we need to cleanup the pubsub and unsubscribe, or does that not much matter? I do very little async coding, so might not even be a valid question